models/vrp.rb

```diff
@@ -188,6 +189,24 @@ def self.check_consistency(hash)
       raise OptimizerWrapper::DiscordantProblemError, 'Rests are not available with periodic heuristic.' unless hash[:vehicles].all?{ |vehicle| vehicle[:rests].to_a.empty? }
     end
 
+    def self.deduce_data_from_hash(hash)
+      return hash if hash[:shipments].to_a.empty?
+
```

Comment:
```suggestion
```
It is not necessary to check if it is empty because if the `hash[:shipments]` is empty/nil then the for loop does nothing.